packages/atomic/src/components/search/atomic-generated-answer/e2e/atomic-generated-answer.e2e.ts

```diff
@@ -1,51 +1,140 @@
+          } else {
+            expect(href).not.toMatch(/#:~:text=/);
+          }
+        }
+      });
     });
 
-    test('should render citation when available', async ({generatedAnswer}) => {
-      const citationCount = await generatedAnswer.getCitationCount();
-      expect(citationCount).toBeGreaterThan(0);
-      await generatedAnswer.citation.first().screenshot();
+    test.describe('with citation anchoring disabled', () => {
+      test.beforeEach(async ({generatedAnswer}) => {
+        await generatedAnswer.load({story: 'disable-citation-anchoring'});
+      });
+
+      test('should not have text fragment when citation anchoring is disabled', async ({
+        generatedAnswer,
+      }) => {
+        await generatedAnswer.waitForCitations();
+        const href = await generatedAnswer.getCitationHref(0);
+
+        expect(href).toBeTruthy();
+        expect(href).not.toContain('#:~:text=');
+      });
     });
+  });
 
-    test('should only append text fragment to HTML citations', async ({
+  test.describe('answer rendering', () => {
+    test('should render answer with bullet points', async ({
       generatedAnswer,
     }) => {
-      const citationCount = await generatedAnswer.getCitationCount();
+      await generatedAnswer.load({story: 'answer-with-bullet-points'});
 
-      for (let i = 0; i < citationCount; i++) {
-        const filetype = await generatedAnswer.getCitationFiletype(i);
-        const href = await generatedAnswer.getCitationHref(i);
+      await generatedAnswer.waitForListItems();
+      const listItems = await generatedAnswer.getListItemTexts();
 
-        expect(href).toBeTruthy();
+      const expectedList = [
+        'Eastern White Pine (Pinus strobus)',
+        'Red Pine (Pinus resinosa)',
+        'Sugar Pine (Pinus lambertiana)',
+        'Scots Pine (Pinus sylvestris)',
+      ];
 
-        if (filetype === 'html') {
-          expect(href).toMatch(/#:~:text=/);
-        } else {
-          expect(href).not.toMatch(/#:~:text=/);
-        }
+      for (const expectedItem of expectedList) {
+        const foundItem = listItems.find((item) => item.includes(expectedItem));
+        expect(foundItem).toBeTruthy();
       }
     });
-  });
 
-  test.describe('with citation anchoring disabled', () => {
-    test.beforeEach(async ({generatedAnswer}) => {
-      await generatedAnswer.load({story: 'disable-citation-anchoring'});
+    test('should render answer with code blocks', async ({generatedAnswer}) => {
+      await generatedAnswer.load({story: 'answer-with-code'});
+
+      await generatedAnswer.waitForCodeBlocks();
+      const codeBlockCount = await generatedAnswer.getCodeBlockCount();
+      expect(codeBlockCount).toBeGreaterThan(0);
+
+      const codeTexts = await generatedAnswer.getCodeBlockTexts();
+      const combinedCode = codeTexts.join(' ');
+      expect(combinedCode).toContain('await window.loadAtomic()');
+      expect(combinedCode).toContain('atomic-commerce-interface');
```

Comment:
I :suspect: at first, thinking we were runnign assertions on "live" JS in a script tag.
But not it's a sample tag code tab displayed in the generated answer 😆